modules/deploy/pages/deployment-option/cloud/terraform-provider.adoc

```diff
@@ -0,0 +1,101 @@
+= Redpanda Terraform Provider
+:description: Learn how to use the Redpanda Terraform provider to create and manage resources in Redpanda Cloud.
+:page-cloud: true
+:page-beta: true
+
+The https://registry.terraform.io/providers/redpanda-data/redpanda/latest[Redpanda Terraform provider^] interfaces with the Cloud API, so you can define your Redpanda infrastructure in Terraform configuration files. You can use the Terraform provider to create and manage the following resources in Redpanda Cloud:
+
+* ACLs
+* Clusters 
+* Networks
+* Resource groups
+* Topics
+* Users
+
+== Prerequisites
+
+. Install Terraform following the https://learn.hashicorp.com/tutorials/terraform/install-cli[Terraform documentation^].
+. Clone the https://github.com/redpanda-data/terraform-provider-redpanda[`terraform-provider-redpanda` Github repository^]:
++
+[,bash]
+----
+git clone https://github.com/redpanda-data/terraform-provider-redpanda.git
+----
+
+== Limitations
+
+*?? TO REVIEWERS: Can you please confirm that the provider is supported on Serverless and Azure? And what functionality is supported in Cloud API but not provider, like Connectors?*
+
+== Configure the provider
+
+. To install the Redpanda provider, copy and paste the following code into your Terraform configuration:
++
+```
+terraform {
+  required_providers {
+    redpanda = {
+      source  = "redpanda-data/redpanda"
+      version = "~> 0.5.1"
+    }
+  }
+}
+```
+
+. Initialize Terraform:
++
+```
+terraform init
+```
+
+. For the Redpanda Terraform provider to authenticate against Redpanda Cloud, enter your `client_id` and `client_secret` credentials in your Terraform configuration file or set them as environment variables:
++
+[tabs]
+======
+Static credentials::
++
+--
+
+```
+provider "redpanda" {
+  client_id      = "<client_id>"
+  client_secret  = "<client_secret>"
+}
+```
+
+--
+Environment variables::
++
+--
+
+```
+REDPANDA_CLIENT_ID=<client_id>
+REDPANDA_CLIENT_SECRET=<client_secret>
+```
+
+--
+======
+
+. *?? TO REVIEWERS: Should we include everything in the README, including https://github.com/redpanda-data/terraform-provider-redpanda/blob/main/README.md#developing-the-provider[Develop the provider] in this doc, or just link to the README?* 
+
+== Use the provider
+
+You can use the Terraform provider to manage data resources in existing Redpanda Cloud clusters. You must use data sources and reference the existing cluster using the cluster ID (Redpanda ID). 
+
+For example, to create a topic in a cluster with ID `foo-bar-baz`:
+
+```
+data "redpanda_cluster" "test" {
+  id = "foo-bar-baz"
+}
+
+resource "redpanda_topic" "test" {
+  name               = "myTopicName"
+  partition_count    = 3
+  replication_factor = 3
+  cluster_api_url    = data.redpanda_cluster.test.cluster_api_url
+}
+```
```

Comment:
This part of the document is about how to use the provider **on an existing** cloud cluster, I think is worth having an additional section on top on how to create your own cluster from 0 using the Terraform Provider. (You can see this example: https://github.com/redpanda-data/terraform-provider-redpanda/blob/main/examples/cluster/aws/main.tf) or maybe point users to it

We have more examples in https://github.com/redpanda-data/terraform-provider-redpanda/tree/main/examples, and it might be worth having a link to them

Reply:
Thank you, @r-vasquez!